lib/setupmanagers/qemuhck/devices/virtio-mem-pci.json

```diff
@@ -0,0 +1,8 @@
+{
+  "name": "virtio-mem-pci",
+  "memory_extra_param": ",maxmem=@double_memory@",
```

Comment:
Why don't you always add `maxmem` and instead make it specific to virtio-mem-pci?

Reply:
In the case of NetKVM (3 VM), we will have the sum of maxmem greater than the host RAM on CI servers. I want to be sure that the host always has some free memory.

Reply:
Is it problematic?

Reply:
I think it's not that strange that the sum of maxmem is greater than the host RAM.

Think of a multi-tenant machine farm; there are multiple machines, and each machine hosts multiple VMs. Also, assume that you want to scale VMs bigger without downtime.
In such a scenario, you set `maxmem` of all machines to match the maximum memory size a VM can hold. The sum of `maxmem` of VMs in a machine can surpass the physical memory size. Later, if too many memory is requested for a machine, some VMs on the machine will migrate to make space.
I think some cloud vendors already do this kind of trick.

Reply:
I don't know the QEMU source related to `maxmem` option. If you can confirm that QEMU will not allocate more than memory_size on Windows Guest without user action. I am ok.

Reply:
You can assume that; I believe the very purpose of `maxmem` is to support a scenario I have just described, and in that case, we should never have a problem with overcomitted `maxmem`.

Reply:
Ok. Let's assume that. We can always change this if see any problem.